lib/runner.js

```diff
@@ -657,6 +661,30 @@ Runner.prototype.abort = function(){
 };
 
 /**
+ * Filter suites based on `isOnly` logic.
+ *
+ * @param {Array} suite
+ * @returns {Array}
+ * @api private
+ */
+function filterOnly(suite) {
+  // If it has `only` tests
+  if(suite.hasOnly) suite.tests = suite.hasOnly;
+  // If it doesn't has a suites make it running
+  // only if it mark as an `only`
+  if(!suite.suites.length) return suite.isOnly;
+  // Filter the nested suites
+  suite.suites = filter(suite.suites, filterOnly);
+  // `Unrunning` suites that not mark as an `only`
```

Comment:
Maybe:
```javascript
//  Don't run suites that are not marked with `only`
```
Though if we are commenting the following line, maybe it would me more adequate:
```javascript
// Don't run tests from suites that are not marked as `only`
```